spirv_hlsl.cpp

```diff
@@ -1273,9 +1273,119 @@ void CompilerHLSL::emit_uniform(const SPIRVariable &var)
 {
 	add_resource_name(var.self);
 	auto &type = get<SPIRType>(var.basetype);
-	if (options.shader_model >= 40 && (type.basetype == SPIRType::Image || type.basetype == SPIRType::SampledImage))
+	if (options.shader_model >= 40 && type.basetype == SPIRType::SampledImage)
 	{
-		statement("Texture2D<float4> ", to_name(var.self), ";");
+		string format = "float4";
```

Comment:
o_O, guess you can throw if regular sampling is attempted on uint/int textures, texelFetch should work though.